darts/anomaly_detection/anomaly_model.py

```diff
@@ -0,0 +1,357 @@
+        """
+        super().__init__()
+
+        raise_if_not(
+            isinstance(model, ForecastingModel),
+            f"Model must be a darts.models.forecasting not a {type(model)}",
+        )
+        self.model = model
+
+        if isinstance(scorer, Sequence):
+            self.scorers = scorer
+        else:
+            self.scorers = [scorer]
+
+        for scorer in self.scorers:
+            raise_if_not(
+                isinstance(scorer, Scorer),
+                "Scorer must be a darts.anomaly_detection.score not a {}".format(
+                    type(scorer)
+                ),
+            )
+
+    def fit(
+        self,
+        series: TimeSeries,
+        model_fit_params: Optional[Dict[str, Any]] = None,
+        hist_forecasts_params: Optional[Dict[str, Any]] = None,
+    ):
+        """Train the model and the scorer(s) on the given time series.
+
+        Parameters
+        ----------
+        series : Darts TimeSeries
+        model_fit_params: dict, optional
+            parameters of the Darts `.fit()` forecasting model
+        hist_forecasts_params: dict, optional
+            parameters of the Darts `.historical_forecasts()` forecasting model
+
+        Returns
+        -------
+        self
+            Fitted Anomaly model (forecasting model and scorer(s))
+        """
+
+        if model_fit_params is None:
+            model_fit_params = {}
+
+        if hist_forecasts_params is None:
+            hist_forecasts_params = {}
+
+        # fit forecasting model
+        if hasattr(self.model, "fit"):
+            if not self.model._fit_called:
+                self.model.fit(series, **model_fit_params)
+
+        # fit scorer model
+        for scorer in self.scorers:
+            if hasattr(scorer, "fit"):
+                pred = self.model.historical_forecasts(
+                    series, retrain=False, **hist_forecasts_params
+                )
+
+                scorer.fit(pred, series)
+
+    def score(
+        self, series: TimeSeries, hist_forecasts_params: Optional[Dict[str, Any]] = None
+    ):
+        """Predicts the given input time series with the forecasting model, and applies the scorer(s)
+        on the prediction and the given input time series. Outputs the anomaly score of the given
+        input time series.
+
+        Parameters
+        ----------
+        series : Darts TimeSeries
+        hist_forecasts_params: dict, optional
+            parameters of the Darts `.historical_forecasts()` forecasting model
+
+        Returns
+        -------
+        Darts TimeSeries
+            Anomaly score time series
+        """
+
+        if hist_forecasts_params is None:
+            hist_forecasts_params = {}
+
+        raise_if_not(
+            self.model._fit_called,
+            f"Model {self.model} has not been trained. Please call .fit()",
+        )
+
+        pred = self.model.historical_forecasts(
+            series, retrain=False, **hist_forecasts_params
+        )
```

Comment:
Here we can't input any parameters for hist_forecasts_params, as if we choose `last_points_only=False` we will have a list of TimeSeries and not just a TimeSeries. The slice_intersect will not work in this case.